src/components/DsrDemo.js

```diff
@@ -2,36 +2,60 @@ import React from 'react';
 import { Flex, Card, Text, Button } from 'rimble-ui';
 import { MDAI } from '@makerdao/dai-plugin-mcd';
 import { approveProxyInDai } from '../utils/web3';
+import BigNumber from "bignumber.js"
+
 
 
 class DsrDemo extends React.Component {
     state = {
         ETH: "0.00 ETH",
-        MDAI: "0.00 MDAI",
-        DSR: "0.00 MDAI",
+        MDAI: 0,
+        DSR: 0,
+        DSRCalc: 0,
+        APR: 0,
+        deposit: 10,
+        time: 1,
+        total: 0
     }
 
+
     componentWillMount() {
         this.displayBalances();
         this.updateBalance();
+        this.getYearlyRate();
     }
 
-
     displayBalances = async () => {
         let maker = this.props.maker;
         let ethBalance = await maker.getToken('ETH').balanceOf(maker.currentAddress())
         let MDAIBalance = await maker.getToken('MDAI').balanceOf(maker.currentAddress());
         let dsrManager = await maker.service('mcd:savings')
         let dsrBalance = await dsrManager.balance();
-        this.setState({ ETH: ethBalance.toString(), MDAI: MDAIBalance.toString(), DSR: dsrBalance.toString()});
+        this.setState({ ETH: ethBalance.toString(), MDAI: BigNumber(MDAIBalance._amount).toNumber(), DSR: BigNumber(dsrBalance._amount).toNumber() });
+    }
+
+    calculateInterest = async () => {
+        let APRperSecond = 1 + (this.state.APR / 365 / 86400) * 15
+        let timeOverBlockCreationTime = this.state.time / 15
+        let total = (Math.pow(APRperSecond, timeOverBlockCreationTime)) * this.state.DSR
+        if (total !== 0) {
+            this.setState({ time: this.state.time + 1000 })
+            this.setState({ total: total })
+        } else {
+            this.setState({ total: 0 })
+        }
     }
 
     updateBalance = async () => {
         setInterval(async () => {
             this.displayBalances()
         }, 5000)
```

Comment:
@ernestpascual this all looks great nice work! additionally, i played with the `ms` param of the `setInterval` func and made it 10 just for some fun rapid updates to show their money is growing. one thing i noticed is that when one setInterval func is set with different `ms` param (one is greater than the other), both setInterval funcs fire following the func w the greater `ms` interval. i guess that's cause they're running on a single thread somehow